crates/js-component-bindgen-component/wit/js-component-bindgen.wit

```diff
@@ -62,6 +62,24 @@ world js-component-bindgen {
     /// Whether to output core Wasm utilizing multi-memory or to polyfill
     /// this handling.
     multi-memory: option<bool>,
+
+    /// Configure whether to use `async` imports or exports with
+    /// JavaScript Promise Integration (JSPI) or Asyncify.
+    async-mode: option<async-mode>,
+  }
+
+  record async-imports-exports {
+    imports: list<string>,
+    exports: list<string>,
+  }
```

Comment:
Since components can contain multiple sub-components, this is currently joining imports and exports across all sub-components.

If we only consider the outer component import and exports, then we should restrict to that, but we can't do that either since internal components must be async if the outer component is!

**Therefore I think we should probably restrict this to outer component imports only**

And we should make exports all async as soon as we go into asyncify mode. This also will then help avoid the easy footgun of inadvertantly not making an export async, and then finding it calls an async import down the line which would trap.